pilot/pkg/networking/apigen/apigen.go

```diff
@@ -142,7 +142,7 @@ func (g *APIGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w *
 		svcs := push.Services(proxy)
 		for _, s := range svcs {
 			// Ignore services that are result of conversion from ServiceEntry.
-			if s.Attributes.ServiceRegistry == serviceregistry.External {
+			if s.Attributes.ServiceRegistry == string(serviceregistry.External) {
```

Comment:
I think let `s.Attributes.ServiceRegistry` to be `ProviderID` type is more better?

Reply:
good suggestion

Reply:
thanks, fixed.